components/sup/src/http_gateway.rs

```diff
@@ -168,27 +218,88 @@ impl Server {
                 Err(_) => HTTP_THREAD_COUNT,
             };
 
-            let bind = server::new(move || {
+            let server = server::new(move || {
                 let app_state = AppState::new(gateway_state.clone());
-                App::with_state(app_state).configure(routes)
-            }).workers(thread_count)
-            .bind(listen_addr.to_string());
+                App::with_state(app_state)
+                    .middleware(Authentication)
+                    .configure(routes)
+            }).workers(thread_count);
+
+            let mut tls_status: Option<ServerStartup> = None;
+            let bind;
+
+            if cert_file.is_some() && key_file.is_some() {
+                let mut config = ServerConfig::new(NoClientAuth::new());
```

Comment:
Minor, but I wonder if we want to default to preferring server cert ordering, I think it is false by default in this library.

Reply:
My TLS knowledge is a bit weak.  Can you elaborate some on this?

Reply:
I'm referring to this option:

https://docs.rs/rustls/0.14.0/rustls/struct.ServerConfig.html#structfield.ignore_client_order

The upshot is that regardless of the client's cipher preferences, we will choose the best available cipher according to the server's preferences.  This is often preferred on the server side to make sure we are using ciphers based on the /servers/ performance and security requirements as opposed to j. random client.

In this case the server only supports ciphers that are all pretty good:

https://docs.rs/rustls/0.14.0/rustls/static.ALL_CIPHERSUITES.html

so it is a pretty minor issue.  But a number of TLS-quality scanners end up looking for it, so it also helps for people who want to do automate scanning of all their TLS listeners.

Reply:
Thank you for the thorough explanation.  It doesn't sound like there's a downside to doing that, so I'll make it so.